File: python/00_session_authors.py
import requests
from lxml.html import fromstring

def make_request():
    url = 'https://2022.drupalcamp.es/programa'
    response = requests.get(url)

    if not response.ok:
        raise Exception

    return response.text

def extract_authors(html):
    dom = fromstring(html)
    author_xpath = '//div[@class="schedule-wrapper"]//span[@class="author"]'
    
    authors = [e.text_content() for e in dom.xpath(author_xpath)]
    return authors 

html = make_request()
authors = extract_authors(html)

print(authors)

File: javascript/00_session_authors.js
const https = require('https')
const jsdom = require("jsdom")
const { JSDOM } = jsdom

const options = {
  hostname: '2022.drupalcamp.es',
  port: 443,
  path: '/programa',
  method: 'GET'
}

const req = https.request(options, res => {
  console.log(`statusCode: ${res.statusCode}`)

  res.on('data', html => {
    const dom = new JSDOM(html).window.document

    const author_elements = dom.querySelectorAll(".author")
    const authors = Array.from(author_elements).map(author => author.textContent) 

    console.log(authors)
  })
})

req.on('error', error => {
    console.error(error)
})

req.end()
